posts/inside-rust/2022-07-29-compiler-team-2022-midyear-report.md

```diff
@@ -0,0 +1,800 @@
+<tr>                                                       <td style=color:darkblue>                      </td><td style=color:darkblue>getting-feedback      </td><td style=color:darkblue>                      </td><td style=color:darkblue>                      </td><td style=color:darkblue>getting-feedback      </td><td style=color:darkblue>                      </td><td style=color:darkblue>                      </td><td style=color:darkblue>                      </td><td style=color:darkblue>                      </td><td style=color:darkblue>                      </td><td style=color:darkblue>                      </td><td style=color:darkblue>getting-feedback      </td><td style=color:darkblue>                      </td><td style=color:darkblue>getting-feedback      </td><td style=color:darkblue>                      </td><td style=color:darkblue>                      </td><td style=color:darkblue>                      </td><td style=color:darkblue>                      </td><td style=color:darkblue>                      </td><td style=color:darkblue>                      </td><td style=color:darkblue>                      </td><td style=color:darkblue>                      </td></tr>
+
+</table>
+
+Here are some trends I noted from looking at the table:
+
+The overwhelming majority of our ambitions, 19 out of 22, reported *some*
+amount of progress. That's great, especially given that several of those
+ambitions were mere aspirations that explicitly had no resources attached to
+them at the start of the year.
+
+Less than a quarter of the ambitions, 6 out of 22, said that their overall
+problem would be resolved, or the bulk of the problem addressed, within the next
+six months. Ten others said they predicted the important components of their
+solution would be available within the next two years, so that's a total of 16
+out of 22 that expect some kind of solution within two years.
+
+  * Note: One respondent noted that the two year threshold embedded in the
+    survey was an "odd timeframe." I do not disagree with that remark. The
+    reality is that I was trying to find a happy medium between two extremes:
+    short timeframes don't allow enough time for significant work to be
+    accomplished, while excessively long timeframes (e.g. greater than five
+    years) are very difficult to predict realistically. But the two year
+    time frame was drawn somewhat out of a hat.
+
+Of the nineteen ambitions that reported some progress so far this year, fourteen
+of them (14/19) also said that at the start of the year, they had either *no
+goals* or *no plan* to achieve those goals. Ten of those fourteen say they now
+have a better understanding of the problem, and five of those fourteen say they
+now have a plan.
+
+  * Here is a question I want to put forward to our community: Do we, as a
```

Comment:
How should we collect responses? Also, which community are we asking? The community of compiler contributors, project contributors, Rust users, etc.

Reply:
The "how should we collect responses" is a good point.

I guess I'm asking the community of compiler contributors.

Reply:
I think that's a good topic for a steering meeting then. As it is, it's likely this will get read by a much larger audience than just the compiler contributors and the post sounds as if we want feedback from the whole world 😄

Reply:
Hmm. Okay. I still want to make it clear in the text that this is a question that I want to be asking, but I will revise the text to reframe the question as one that I will be trying to resolve in collaboration with the compiler-contributors.